sys/shell/cmds/gnrc_icmpv6_echo.c

```diff
@@ -207,7 +208,13 @@ static int _configure(int argc, char **argv, _ping_data_t *data)
                     /* intentionally falls through */
                 case 's':
                     if ((++i) < argc) {
-                        data->datalen = atoi(argv[i]);
+                        value = atoi(argv[i]);
+
+                        if ((value < 0) || ((unsigned)value > (UINT16_MAX - sizeof(icmpv6_hdr_t)))) {
+                            printf("ICMPv6 datagram size should be in range 0-65527.\n");
```

Comment:
tbh I don't think you need to check that twice. You already have that check in `gnrc_icmpv6_echo_send()`, no need to pre-check - you can just let `gnrc_icmpv6_echo_send()` return error.

Sending ICMP messages larger than 65527 bytes is not such a common use-case that we need special handling for it in the shell command.

Reply:
There is nothing speaking against pre-checking if you want to have user-friendly error messages, IMHO. At least the `value < 0` should stay in any case, since `value` is `int`, but `len` is `size_t` (so unsigned). Casting could introduce some unexpected values here (especially if `size_t` is set to `uint16_t` by the architecture).

Reply:
I agree that now I have to checks. The one in the API - protects users which would like send icmp pings in the own code.  The second check as @miri64  write gives more user friendly warning. Instead, user when gives wrong size see only cryptic text:

```
All up, running the shell now
> ping -s -7 ::1 
ping -s -7 ::1
error: -22
error: -22
error: -22

--- ::1 PING statistics ---
3 packets transmitted, 0 packets received, 100% packet loss
>
```